pwnlib/tubes/process.py

```diff
@@ -587,10 +593,15 @@ def _validate(self, cwd, executable, argv, env):
 
         # If there's no path component, it's in $PATH or relative to the
         # target directory.
+        # Try all of the file extensions in $PATHEXT on Windows too.
         #
         # For example, 'sh'
-        elif os.path.sep not in executable and which(executable, path=path):
-            executable = which(executable, path=path)
+        elif os.path.sep not in executable and any(which(executable + pathext, path=path) for pathext in pathexts):
```

Comment:
I think this should be moved into `which` itself, as other parts of pwntools might benefit from it.